pkgs/development/libraries/agda/generic/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib, mkDerivation, fetchFromGitHub, standard-library }:
+
+mkDerivation rec {
+  pname = "generic";
+  version = "0.1";
+
+  src = fetchFromGitHub {
+    repo = "Generic";
+    owner = "effectfully";
+    rev = "v${version}";
+    sha256 = "121121rg3daaqp91845fbyws6g28hyj1ywmh12n54r3nicb35g5q";
+  };
+
+  buildInputs = [
+    standard-library
+  ];
+
+  preBuild = ''
```

Comment:
It would be really cool if this could be abstracted into the agdaPackages lib. Either as simple option `generateEverythingFile = true;` or a transformation `withEverythingFile (fetchFromGitHub { ... })` on the source.

Reply:
I agree but maybe this should a different PR. A potential problem with this is that you need a way to get where the source lives. This is given as the `include` field in the `.agda-lib` file but we don't currently extract it. This file could also contain multiple include sources

Reply:
This should potentially be in `preConfigure` instead of `preBuild` but as `agda` packages don't really have a configure stage, this doesn't really matter

Reply:
Ok, so let's touch this again when there is another library where this trick comes in handy.